syne_tune/backend/sagemaker_backend/sagemaker_backend.py

```diff
@@ -242,6 +283,58 @@ def _resume_trial(self, trial_id: int):
         else:
             self.resumed_counter[trial_id] = 1
 
+    def busy_trial_ids(self) -> List[Tuple[int, str]]:
+        if self._busy_trial_ids:
+            trial_ids_and_names = [
+                (trial_id, self.job_id_mapping[trial_id])
+                for trial_id in self._busy_trial_ids
+            ]
+            trial_results = sagemaker_search(
+                trial_ids_and_names, sm_client=self.sm_client
+            )
+            busy_list = []
+            new_busy_trial_ids = set()
+            reported_trial_ids = set()
+            for result in trial_results:
+                trial_id = result.trial_id
+                status = result.status
+                reported_trial_ids.add(trial_id)
+                if status in BUSY_STATUS:
+                    busy_list.append((trial_id, result.status))
+                    new_busy_trial_ids.add(trial_id)
+                    if (
+                        status == Status.stopping
+                        and trial_id not in self._stopping_time
+                    ):
+                        # First time we see `Status.stopping` for this `trial_id`
+                        self._stopping_time[trial_id] = time.time()
+                elif trial_id in self._stopping_time:
+                    # Trial just stopped being busy
+                    stop_time = time.time() - self._stopping_time[trial_id]
+                    logger.info(
+                        f"Stopping delay for trial_id {trial_id}: {stop_time:.2f} secs"
+                    )
+                    del self._stopping_time[trial_id]
```

Comment:
This logic is (more or less) a duplicate of lines 161-170. I think we should extract this functionality to a separate function.

Reply:
I'll rather remove the duplicate